include/nil/crypto3/zk/math/expression.hpp

```diff
@@ -338,7 +341,7 @@ namespace nil {
                 binary_arithmetic_operation(
                         const expression<VariableType>& expr_left,
                         const expression<VariableType>& expr_right,
-                        ArithmeticOperator op) 
+                        ArithmeticOperator op)
                     : expr_left(expr_left)
                     , expr_right(expr_right)
                     , op(op) {
```

Comment:
Please add  update_hash(); here, and remove the other calls to update_hash you have added, they are not needed.

Reply:
Done!